portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/BullhornServiceImpl.java

```diff
@@ -311,6 +320,57 @@ public void update(Observable o, final Object arg) {
                                 switchToNull();
                                 lock(sa);
                             }
+                        } else if (LessonBuilderEvents.COMMENT_CREATE.equals(event)) {
+                            try {
+                                long commentId = Long.parseLong(pathParts[pathParts.length - 1]);
+                                SimplePageComment comment = simplePageToolDao.findCommentById(commentId);
+
+                                // We have to do this to get the item id of the page hosting the comments item. The
+                                // itemId parameter is literally this id.
+                                List<SimplePageItem> pageItems = simplePageToolDao.findPageItemsBySakaiId(Long.toString(comment.getPageId()));
+                                long pageItemId = pageItems.get(0).getId();
+
+                                SimplePage page = simplePageToolDao.getPage(comment.getPageId());
+                                SitePage sitePage = siteService.findPage(page.getToolId());
+
+                                String toolId = sitePage.getTools(new String[] {"sakai.lessonbuildertool"}).iterator().next().getId();
+                                String siteId = page.getSiteId();
+
+                                String url = serverConfigurationService.getPortalUrl() + "/site/" + siteId
```

Comment:
I'd be happier if this we in the lessons API so that is seen the responsibility of the tool to build the URL and we don't start breaking URLs when someone adds a new required parameter.